src/qml/qgismobileapp.qml

```diff
@@ -1397,6 +1399,122 @@ ApplicationWindow {
+        visible: dashBoard.activeLayer.geometryType() === Qgis.GeometryType.Polygon
+          || dashBoard.activeLayer.geometryType() === Qgis.GeometryType.Line
+        iconSource: Theme.getThemeVectorIcon( "ic_common_angle_white_24dp" )
+        iconColor: "white"
+        bgcolor: Theme.darkGray
+
+        property bool isSnapToCommonAngleEnabled: false
+        property bool isSnapToCommonAngleRelative: false
+        property int snapToCommonAngleDegrees: 45
+
+        state: isSnapToCommonAngleEnabled ? "On" : "Off"
+
+        states: [
+          State {
+            name: "Off"
+            PropertyChanges {
+              target: snapToCommonAngleButton
+              iconColor: "white"
+              bgcolor: Theme.darkGraySemiOpaque
+            }
+          },
+
+          State {
+            name: "On"
+            PropertyChanges {
+              target: snapToCommonAngleButton
+              iconColor: Theme.mainColor
+              bgcolor: Theme.darkGray
+            }
+          }
+        ]
+
+        onClicked: {
+          isSnapToCommonAngleEnabled = !isSnapToCommonAngleEnabled
+
+          displayToast( isSnapToCommonAngleEnabled ? qsTr( "Snap to %1° angle turned on" ).arg( snapToCommonAngleDegrees ) : qsTr( "Snap to common angle turned off" ) );
+        }
+
+        onPressAndHold: {
+          snapToCommonAngleMenu.popup( parent.x, parent.y )
+        }
+
+        onIsSnapToCommonAngleEnabledChanged: {
+          settings.setValue( "/QField/Digitizing/SnapToCommonAngleIsEnabled", isSnapToCommonAngleEnabled );
+        }
+
+        onIsSnapToCommonAngleRelativeChanged: {
+          settings.setValue( "/QField/Digitizing/SnapToCommonAngleIsRelative", isSnapToCommonAngleRelative );
+        }
+
+        onSnapToCommonAngleDegreesChanged: {
+          settings.setValue( "/QField/Digitizing/SnapToCommonAngleDegrees", snapToCommonAngleDegrees );
+        }
+
+        Component.onCompleted: {
+          isSnapToCommonAngleEnabled = settings.valueBool( "/QField/Digitizing/SnapToCommonAngleIsEnabled", false );
+          isSnapToCommonAngleRelative = settings.valueBool( "/QField/Digitizing/SnapToCommonAngleIsRelative", false );
+          snapToCommonAngleDegrees = settings.valueInt( "/QField/Digitizing/SnapToCommonAngleDegrees", snapToCommonAngleDegrees );
+        }
+
+
+        Menu {
+            id: snapToCommonAngleMenu
+
+            MenuItem {
+              text: qsTr( "Relative angle" )
+              font: Theme.defaultFont
+              height: 48
+              leftPadding: 10
+
+              checkable: true
+              checked: snapToCommonAngleButton.isSnapToCommonAngleRelative
+              onCheckedChanged: {
+                snapToCommonAngleButton.isSnapToCommonAngleRelative = checked
+              }
+            }
+
+            MenuSeparator { width: parent.width }
+
+            Repeater {
+                // list of common angles to snap to
+                model: [10, 15, 30, 45]
```

Comment:
I think we'd benefit from having a 90 degrees item in there too.

Reply:
The reason why I didn't add 90 deg is because it's easily achievable with 45 is already quite obvious when you step from 45 to 90 deg snap. Will add it.